File: clang/test/Driver/response-file.c
// Check that clang is able to process short response files
// Since this is a short response file, clang must not use a response file
// to pass its parameters to other tools. This is only necessary for a large
// number of parameters.
// RUN: echo "-DTEST" > %t.0.txt
// RUN: %clang -E @%t.0.txt %s -v 2>&1 | FileCheck %s -check-prefix=SHORT
// SHORT-NOT: Arguments passed via response file
// SHORT: extern int it_works;

// Check that clang is able to process long response files, routing a long
// sequence of arguments to other tools by using response files as well.
// We generate a 2MB response file to attempt to surpass any system limit.
// But there's no guarantee that we actually will (the system limit could be
// *huge*), so just check that invoking cc1 succeeds under these conditions.
//
// RUN: %clang -E %S/Inputs/gen-response.c | grep DTEST > %t.1.txt
// RUN: %clang -E @%t.1.txt %s -v 2>&1 | FileCheck %s -check-prefix=LONG
// LONG: extern int it_works;

#ifdef TEST
extern int it_works;
#endif

File: flang/test/Driver/response-file.f90
! Test that the driver can process response files.

! RUN: echo "-DTEST" > %basename_t.rsp
! RUN: %flang -E -cpp @%basename_t.rsp %s -o - | FileCheck %s
! RUN: %flang_fc1 -E -cpp @%basename_t.rsp %s -o - | FileCheck %s
! RUN: not %flang %basename_t.rsp %s -o /dev/null
! RUN: not %flang_fc1 %basenamt_t.rsp %s -o /dev/null

! CHECK-LABEL: program test
! CHECK: end program

#ifdef TEST
program test
end program
#else
We should have read the define from the response file.
#endif
